### ISSUE TITLE
ModelFactory and pagination

### ISSUE THREAD

**Issue by juristr** [Swimlane/angular-model-factory#28]
Hi,

how would you implement pagination with modelFactory? My api currently works like this

```
// call
GET /api/v1/people?page=1&rows=3

// result
{
    rows: [
        { id: 1, ... },
        { id: 2, ... },
        ...
    ],
    numberPages: 5,
    totalNumRecords: 13,
    currentPage: 1
}
```

I'm calling the api with `PersonModel.query({ page: 2, numRows: 5 })`. This however results in an error as ModelFactory expects an array as the result.

How would you implement this?? Because the "real PersonModel" entities are within the `rows` property, while the others simply represent the metadata for the pagination on the client.

@amcdnl Any ideas on how to use ModelFactory in such scenario or maybe how to extend it? How did you do the pagination?

**Comment by amcdnl** [comment]
I do: `people.query({ limit: 15, offset: 5 })` which would look like: `GET /api/people?limit=15&offset=5`

**Comment by juristr** [comment]
Sure, that's what I do as well. My issue was more the response from the server. When you have a pagination ctrl like this..

![image](https://cloud.githubusercontent.com/assets/542458/6286280/f67652e6-b905-11e4-97f5-23581ebad4a5.png)

...then you may want to get some more info from the server like the number of pages, the total number of records etc.. That's why my response looks like I've shown above. Now, since `query(..)` by default wraps everything in a `Model.List` I had the issue which I then fixed in #29 . So I overwrite the `isArray` on `query()` and then

```javascript
PersonModel.query({ page: currentPage, rows: maxRowsPerPage})
   .then(function(result){
        // extract and manually wrap the models from the server
        vm.personList = new PersonModel.List(result.rows);

        // update paging information with the other props
   }); 
```

That works. Don't really like the necessity to manually wrap the model list...but for now it works and maybe I find a better way of doing it.

**Comment by amcdnl** [comment]
Ah, I see your point! I'm thinking this would be a great spot for an extension.  For example, on the request type you define a extension method and it handles this kinda automagical.

     actions: {  'query': {  extensions: [ Pagination ] } }

**Comment by Ye-Ting** [comment]
I not good at English .

change 
```
params.afterRequest && (response = params.afterRequest(response))  at line 668 
```


And change
```
actions: {
 'query': {
     afterRequest:function(response) {
            return response.rows;
     }
   }
 }
```
then 
```
new Model.List(response)  at line 681
```
will be ok!

**Comment by juristr** [comment]
Solved with PR #41
